sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala

```diff
@@ -129,7 +129,81 @@ case class GpuLocalLimitExec(limit: Int, child: SparkPlan) extends GpuBaseLimitE
  * Take the first `limit` elements of the child's single output partition.
  */
 case class GpuGlobalLimitExec(limit: Int, child: SparkPlan, offset: Int) extends GpuBaseLimitExec {
+  assert(limit >= 0 || (limit == -1 && offset > 0))
   override def requiredChildDistribution: List[Distribution] = AllTuples :: Nil
+
+  override def doExecuteColumnar(): RDD[ColumnarBatch]  = {
+    val opTime = gpuLongMetric(OP_TIME)
+    val childRdd = child.executeColumnar()
+    childRdd.mapPartitions {
+      iter => new Iterator[ColumnarBatch] {
+        private var remainingLimit = limit - offset
+        private var remainingOffset = offset
```

Comment:
No need to cache the offset value, but keep getting batches from children until skip `offset` rows. We return an empty batch iff there are not enough rows. e.g. offset is 100, but this partition has only 50 rows.